### PR TITLE
:fire: delegate nan behavior to aggregators

### PR DESCRIPTION
**Pull request by jonasvdd**
Removed the `check_nans` argument of the FigureResampler constructor and its `add_traces` method. This argument was used to check for NaNs in the input data, but this is now handled by the `nan_policy` argument of specific aggregators (see for instance the constructor of the `MinMax` and `MinMaxLTTB` aggregator).

### CODE REVIEW

**Comment by jonasvdd** [comment]
Ready for review @jvdd!

File: plotly_resampler/aggregation/aggregators.py [hunk 1]
```diff
@@ -17,8 +17,11 @@
     LTTBDownsampler,
     MinMaxDownsampler,
     MinMaxLTTBDownsampler,
+    NaNMinMaxDownsampler,
+    NaNMinMaxLTTBDownsampler,
 )
 
+# TODO -> integrate NANM4 (after the candlestick PR)
```
Comment (jvdd) [inline plotly_resampler/aggregation/aggregators.py:24]: Make an issue of this?

**Review by jvdd — APPROVED** [review]

File: plotly_resampler/aggregation/aggregators.py (hunk 1 above)
Comment (jonasvdd) [reply plotly_resampler/aggregation/aggregators.py:24]: I agree!

**Review by jvdd — APPROVED** [review]
NaN support for M4 is still open

**Comment by jonasvdd** [comment]
I added the issue, so this can be merged @jvdd?

**Pull request merged**